lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,28 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+    # converts the include hash to a standard format
+    # for constrained serialization recursion
+    #
+    # converts
+    #  [:author, :comments => [:author]] to
+    #  {:author => {}, :comments => { author: {} } }
+    #
+    # and
+    #  [:author, :comments => {:author => :bio}, :posts => [:comments]] to
+    #  {:author => {}, :comments => { author: { bio: {} } }, :posts => { comments: {} } }
+    #
+    # The data passed in to this method could be an array, hash or symbol
+    #
+    # the point of this method is to normalize the include
+    # options for the child relationships.
+    def include_options_to_hash(included)
```

Comment:
If someone could supply an infinitely nested hash or array, then it would be dangerous. thankfully we have the else clause